.github/workflows/ci.yml

```diff
@@ -89,14 +89,14 @@ jobs:
       uses: dtolnay/rust-toolchain@stable
     - run: echo "VCPKG_ROOT=$env:VCPKG_INSTALLATION_ROOT" | Out-File -FilePath $env:GITHUB_ENV -Append
     - run: vcpkg install openssl:x64-windows-static-md
-    - name: Run cargo check --workspace
-      run: cargo check --workspace
+    - name: Run cargo check --workspace --all-targets
+      run: cargo check --workspace --all-targets
     - name: Run the tests
-      run: cargo test --workspace
+      run: cargo test --workspace --all-targets
```

Comment:
`test` defaults to `--all-targets` so I think we should leave it out here and below.

Reply:
I swore I tested this and didn't see it processing the examples directory but I might have made a mistake. I'll double check and drop it if so

Reply:
Yeah, I'm not sure if there's something deeper going on here, but I see a difference when `--all-targets` isn't provided and I run `cargo test` in the root of the repo:

Without:
```
[daniel@blanc:~/Code/Rust/rcgen]$ cargo test 1>/dev/null
    Finished test [unoptimized + debuginfo] target(s) in 0.05s
     Running unittests src/lib.rs (target/debug/deps/rcgen-f88f7e4d6f3a7bd5)
     Running tests/botan.rs (target/debug/deps/botan-c52964ac1215ee80)
     Running tests/generic.rs (target/debug/deps/generic-19ee0dcd0b8cb3b1)
     Running tests/openssl.rs (target/debug/deps/openssl-0a1d87c1cbad2f2f)
     Running tests/util.rs (target/debug/deps/util-a1c4b7443d444db3)
     Running tests/webpki.rs (target/debug/deps/webpki-94dd7b7973978afc)
     Running unittests src/main.rs (target/debug/deps/rustls_cert_gen-0df5c6c468d810e0)
   Doc-tests rcgen
```

With:
```
[daniel@blanc:~/Code/Rust/rcgen]$ cargo test --all-targets 1>/dev/null
    Finished test [unoptimized + debuginfo] target(s) in 0.06s
     Running unittests src/lib.rs (target/debug/deps/rcgen-f88f7e4d6f3a7bd5)
     Running tests/botan.rs (target/debug/deps/botan-c52964ac1215ee80)
     Running tests/generic.rs (target/debug/deps/generic-19ee0dcd0b8cb3b1)
     Running tests/openssl.rs (target/debug/deps/openssl-0a1d87c1cbad2f2f)
     Running tests/util.rs (target/debug/deps/util-a1c4b7443d444db3)
     Running tests/webpki.rs (target/debug/deps/webpki-94dd7b7973978afc)
     Running unittests examples/rsa-irc.rs (target/debug/examples/rsa_irc-1f3811a2067a25e1)
     Running unittests examples/rsa-irc-openssl.rs (target/debug/examples/rsa_irc_openssl-33576615a0a33f63)
     Running unittests src/main.rs (target/debug/deps/rustls_cert_gen-0df5c6c468d810e0)
```

Notable the `--all-targets` version is the only one that outputs lines for `examples/...`

Reply:
Well, I don't think we need `cargo test` to cover examples as long as we have `cargo check --all-targets` covering examples?

Reply:
Is there harm in having `cargo test` use `--all-targets`? I admit it seems unlikely there will be unit tests added to an example but it seems better to be in a state where if that does happen we don't need to remember to make any changes to CI.

I don't feel strongly but I also don't see the downside.

Reply:
Oh, so this is about unit tests in examples... Sure, I guess it's fine.

Reply:
Yup: my primary goal is to ensure that all code is built in CI, and all tests are run, no matter where they're defined.